gateway/ln-gateway/src/bin/gatewayd.rs

```diff
@@ -40,7 +42,19 @@ pub struct GatewayOpts {
 
     /// Public URL to a Gateway Lightning rpc service
     #[arg(long = "lnrpc-addr", env = "FM_GATEWAY_LIGHTNING_ADDR")]
-    pub lnrpc_addr: Url,
+    pub lnrpc_addr: Option<Url>,
```

Comment:
Why are these options? I see below that if they're none, we just exit with error code 1. Can we make these parameters required?

Reply:
`lnrpc_addr` uses the our custom gRPC spec to talk to an abstract lightning node. Currently that's just the core-lightning plugin, but in the future it could be an integration to some LSP etc.

The `lnd_` parameters are specifically for LND.

The invariant should be that either `lnrpc_addr` is provided, or all the `lnd_` variables are provided, and not both. One or the other. Line 119 checks whether one set or the other is present, but doesn't complain if both are present. But I think it probably should.

Reply:
tracked todo for issue #1955